asmcomp/amd64/emit.mlp

```diff
@@ -685,6 +685,11 @@ let emit_instr fallthrough i =
   | Lop(Iintop_imm(op, n)) ->
       (* We have i.arg.(0) = i.res.(0) *)
       instr_for_intop op (int n) (res i 0)
+  | Lop(Icompf cmp) ->
+      let cond, a0, a1, _ = float_cond_and_swap cmp (arg i 0) (arg i 1) in
+      I.cmpsd cond a1 a0;
+      I.movd a0 (res i 0);
+      I.neg (res i 0);
```

Comment:
Sorry for the very dummy question (showing, btw, that I should really not be the one reviewing this PR...), but why is the negation necessary?

Reply:
compsd's result is either all one bits (-1) or zero, but we actually want zero or one, so neg is the easiest way to do the conversion (and, iirc, the instruction encoding of neg is also pretty short)

Reply:
Thanks!  (It was pretty obvious from https://www.felixcloutier.com/x86/cmpsd but I looked at the wrong pseudo code...)